debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/ShrCCommand.java

```diff
@@ -1042,167 +1134,195 @@ private void dbgShrcPrintAllStats(PrintStream out, J9JavaVMPointer vm, J9SharedC
-		 */
-		if (cacheDescriptor.notNull()) {
-			romclassStartAddress = UDATA.cast(cacheDescriptor.romclassStartAddress());
-		}
-		
-		/*
-		 * If romclassStartAddress could not get on first try, 
-		 * 2nd try : Calculate it manually by adding readWriteBytes to cacheStartAddress.
-		 */
-		if ((0 == romclassStartAddress.longValue()) && (0 != cacheStartAddress.readWriteBytes().longValue())) {
-			romclassStartAddress = new UDATA(cacheStartAddress.addOffset(cacheStartAddress.readWriteBytes()).longValue());
-		}
-		
-		/*
-		 * If romclassStartAddress could not get on the second try, 
-		 * then print info message and continue running.
-		 */
-		if (0 == romclassStartAddress.longValue()) {
-			CommandUtils.dbgPrint(out,"romclassStartAddress is zero");
-		}
-		
-		/*
-		 * TRY TO GET segmentPtr
-		 * 1st try : Try to get it by resolving segmentSRP from cacheStartAddress
-		 */
-		UDATA segmentSRP = cacheStartAddress.segmentSRP();
-		if (0 != segmentSRP.longValue()) {
-			segmentPtr = UDATA.cast(cacheStartAddress).add(segmentSRP);
-		}
+			/* 
+			 * TRY TO GET romclassStartAddress
+			 * 1st try : Get it from cacheDescriptor
+			 */
+			if (cacheDescriptor.notNull()) {
+				romclassStartAddress[layer] = UDATA.cast(cacheDescriptor.romclassStartAddress());
+			}
+			
+			/*
+			 * If romclassStartAddress could not get on first try, 
+			 * 2nd try : Calculate it manually by adding readWriteBytes to cacheStartAddress.
+			 */
+			if ((0 == romclassStartAddress[layer].longValue()) && (0 != cacheStartAddress[layer].readWriteBytes().longValue())) {
+				romclassStartAddress[layer] = new UDATA(cacheStartAddress[layer].addOffset(cacheStartAddress[layer].readWriteBytes()).longValue());
+			}
+			
+			/*
+			 * If romclassStartAddress could not get on the second try, 
+			 * then print info message and continue running.
+			 */
+			if (0 == romclassStartAddress[layer].longValue()) {
+				CommandUtils.dbgPrint(out, "romclassStartAddress is zero for layer %d\n", layer);
+			}
+			
+			/*
+			 * TRY TO GET segmentPtr
+			 * 1st try : Try to get it by resolving segmentSRP from cacheStartAddress
+			 */
+			UDATA segmentSRP = cacheStartAddress[layer].segmentSRP();
+			if (0 != segmentSRP.longValue()) {
+				segmentPtr[layer] = UDATA.cast(cacheStartAddress[layer]).add(segmentSRP);
+			}
 
-		/*
-		 * If segmentPtr could not get on the second try, 
-		 * then print info message and continue running.
-		 * 
-		 */
-		if (0 == segmentPtr.longValue()) {
-			CommandUtils.dbgPrint(out,"segmentPtr is zero");
-		}
+			/*
+			 * If segmentPtr could not get on the second try, 
+			 * then print info message and continue running.
+			 * 
+			 */
+			if (0 == segmentPtr[layer].longValue()) {
+				CommandUtils.dbgPrint(out, "segmentPtr is zero for layer %d\n", layer);
+			}
+			cacheDescriptor = cacheDescriptor.previous();
+			layer += 1;
+		} while (layer <= topLayer);
 		
 		return new ShrcConfig(cacheStartAddress, romclassStartAddress, segmentPtr);
 	}
 
 	class ShrcConfig {
```

Comment:
This class should be `static`.

Reply:
Yes. But I didn't touch this line in the PR. So I will keep it the way it is now.

Reply:
You changed the next line; I think you can change this one too. ;-)